fastly/errors.go

```diff
@@ -11,147 +11,150 @@ import (
+var ErrMissingRuleID = errors.New("missing required field 'RuleID'")
 
 // ErrMissingConfigSetID is an error that is returned was an input struct
 // requires a "ConfigSetID" key, but one was not set
-var ErrMissingConfigSetID = errors.New("Missing required field 'ConfigSetID'")
+var ErrMissingConfigSetID = errors.New("missing required field 'ConfigSetID'")
 
 // ErrMissingWAFList is an error that is returned was an input struct
 // requires a list of WAF id's, but it is empty
 var ErrMissingWAFList = errors.New("WAF slice is empty")
 
 // ErrMissingPool is an error that is returned when an input struct requires
 // a "Pool" key, but one was not set.
-var ErrMissingPool = errors.New("Missing required field 'Pool'")
+var ErrMissingPool = errors.New("missing required field 'Pool'")
 
 // ErrMissingServer is an error that is returned when an input struct requires
 // a "Server" key, but one was not set.
-var ErrMissingServer = errors.New("Missing required field 'Server'")
+var ErrMissingServer = errors.New("missing required field 'Server'")
 
 // ErrMissingAddress is an error that is returned when an input struct requires
 // a "Address" key, but one was not set.
-var ErrMissingAddress = errors.New("Missing required field 'Address'")
+var ErrMissingAddress = errors.New("missing required field 'Address'")
 
 // ErrBatchUpdateMaximumItemsExceeded is an error that indicates that too many batch operations are being executed.
 // The Fastly API specifies an maximum limit.
 var ErrBatchUpdateMaximumOperationsExceeded = errors.New("batch modify maximum operations exceeded")
 
 // ErrMissingKMSKeyID is an error that is returned from an input struct that requires
 // a "ServerSideEncryptionKMSKeyID" key, but one was not set.
-var ErrMissingKMSKeyID = errors.New("Missing required field 'ServerSideEncryptionKMSKeyID'")
+var ErrMissingKMSKeyID = errors.New("missing required field 'ServerSideEncryptionKMSKeyID'")
 
 // ErrMissingCertBlob is an error that is returned from an input struct that requires
 // a "CertBlob" key, but one was not set.
-var ErrMissingCertBlob = errors.New("Missing required field 'CertBlob'")
+var ErrMissingCertBlob = errors.New("missing required field 'CertBlob'")
 
 // ErrMissingIntermediatesBlob is an error that is returned from an input struct that requires
 // a "IntermediatesBlob" key, but one was not set.
-var ErrMissingIntermediatesBlob = errors.New("Missing required field 'IntermediatesBlob'")
+var ErrMissingIntermediatesBlob = errors.New("missing required field 'IntermediatesBlob'")
 
 // ErrStatusNotOk is an error that indicates that indicates that the response body returned
 // by the Fastly API was not `{"status": "ok"}`
-var ErrStatusNotOk = errors.New("Unexpected 'status' field in API response body")
+var ErrStatusNotOk = errors.New("unexpected 'status' field in API response body")
+
+// ErrNotOK is a generic error indicating that something is not okay.
+var ErrNotOK = errors.New("not ok")
```

Comment:
This is new.

Reply:
I'd agree that `ErrNotOK` is the correct variant. And would prefer to go with that.